pkg/controller/completed_workflows.go

```diff
@@ -95,26 +97,25 @@ func CompletedWorkflowsSelectorOutsideRetentionPeriod(retentionPeriodHours int,
 		Operator: v1.LabelSelectorOpNotIn,
 		Values:   hoursToKeep,
 	})
-	s.MatchExpressions = append(s.MatchExpressions, v1.LabelSelectorRequirement{
-		Key:      hourOfDayCompletedKey,
-		Operator: v1.LabelSelectorOpIn,
-		Values:   []string{""},
-	})
+
 	return s
 }
 
-func CompletedWorkflowsSelectorOutsideRetentionPeriodAbandon(retentionPeriodHours int, currentTime time.Time) *v1.LabelSelector {
+// DeprecatedCompletedWorkflowsSelectorOutsideRetentionPeriod
+// Deprecated
+func DeprecatedCompletedWorkflowsSelectorOutsideRetentionPeriod(retentionPeriodHours int, currentTime time.Time) *v1.LabelSelector {
 	hoursToDelete := CalculateHoursToDelete(retentionPeriodHours-1, currentTime.Hour())
 	s := CompletedWorkflowsLabelSelector()
 	s.MatchExpressions = append(s.MatchExpressions, v1.LabelSelectorRequirement{
 		Key:      hourOfDayCompletedKey,
 		Operator: v1.LabelSelectorOpIn,
 		Values:   hoursToDelete,
 	})
+
 	return s
 }
 
-func SplitTimeToMeetsLabelFormat(currentTime time.Time) string {
+func FormatTimeForLabel(currentTime time.Time) string {
 	return strings.ReplaceAll(
 		strings.Split(currentTime.Round(time.Hour).String(), ":")[0], " ", ".")
```

Comment:
Can we make this currentTime.Hour()? It should work fine as is because the same function is used to set the completed label time, but there's something about a completed time of 11:31 being labeled as the hour 12 that doesn't seem right.

Reply:
`currentTime.Hour()` returns an int for the hour within the day... so if you have a retention of 25 hours or something, you wouldn't know whether a label of value "1" has completed today or yesterday... no?

> but there's something about a completed time of 11:31 being labeled as the hour 12 that doesn't seem right.
Similarly, 11:59 will be in the same hour as 11:01.. I think we should just document that to say "deletion will happen +/- an hour of the retention period". 
We can make this more granular (per half hour...) I don't know what's the limit (size of request, perf... etc.) when it comes to the matching expression...

Reply:
Sure, I figured it would be used in on conjunction with the Year(), Month(), and Day() functions to output identical to the current function (ie. "yyyy-mm-dd.hh"). Very nit, OK with not making the change.

Reply:
Agreed... addressed... better?